pkg/controller/atlasbackupcompliancepolicy/atlasbackupcompliancepolicy_controller.go

```diff
@@ -155,18 +155,18 @@ func (r *AtlasBackupCompliancePolicyReconciler) skip(ctx context.Context, log *z
 	return workflow.OK().ReconcileResult()
 }
 
-func (r *AtlasBackupCompliancePolicyReconciler) invalidate(invalid workflow.Result) (ctrl.Result, error) {
+func (r *AtlasBackupCompliancePolicyReconciler) invalidate(invalid workflow.Result) ctrl.Result {
 	// note: ValidateResourceVersion already set the state so we don't have to do it here.
 	r.Log.Debugf("AtlasBackupCompliancePolicy is invalid: %v", invalid)
-	return invalid.ReconcileResult(), nil
+	return invalid.ReconcileResult()
```

Comment:
Although this makes sense from the CI lint perspective, I think it makes it a bit less readable, as these methods are "transitions" from the state-machine perspective. Maybe it worth to disable specific rule in the linter?